classes/decision/types/BackToInternalReviewFromExternalReview.inc.php

```diff
@@ -0,0 +1,233 @@
+    }
+
+    public function validate(array $props, Submission $submission, Context $context, Validator $validator, ?int $reviewRoundId = null)
+    {
+        // If there is no review round id, a validation error will already have been set
+        if (!$reviewRoundId) {
+            return;
+        }
+
+        parent::validate($props, $submission, $context, $validator, $reviewRoundId);
+
+        if (!$this->canRetract($submission, $reviewRoundId)) {
+            $validator->errors()->add('restriction', __('editor.submission.decision.backToInternalReviewFromExternalReviewRound.restriction'));
+        }
+
+        if (!isset($props['actions'])) {
+            return;
+        }
+
+        foreach ((array) $props['actions'] as $index => $action) {
+            $actionErrorKey = 'actions.' . $index;
+            switch ($action['id']) {
+                case $this->ACTION_NOTIFY_AUTHORS:
+                    $this->validateNotifyAuthorsAction($action, $actionErrorKey, $validator, $submission);
+                    break;
+                case $this->ACTION_NOTIFY_REVIEWERS:
+                    $this->validateNotifyReviewersAction($action, $actionErrorKey, $validator, $submission, $reviewRoundId, self::REVIEW_ASSIGNMENT_ACTIVE);
+                    break;
+            }
+        }
+    }
+
+    public function runAdditionalActions(Decision $decision, Submission $submission, User $editor, Context $context, array $actions)
+    {
+        parent::runAdditionalActions($decision, $submission, $editor, $context, $actions);
+
+        foreach ($actions as $action) {
+            switch ($action['id']) {
+                case $this->ACTION_NOTIFY_AUTHORS:
+                    $this->sendAuthorEmail(
+                        new DecisionBackToInternalReviewNotifyAuthor($context, $submission, $decision),
+                        $this->getEmailDataFromAction($action),
+                        $editor,
+                        $submission,
+                        $context
+                    );
+                    break;
+                case $this->ACTION_NOTIFY_REVIEWERS:
+                    $this->sendReviewersEmail(
+                        new DecisionReviewerUnassignedNotifyReviewer($context, $submission, $decision),
+                        $this->getEmailDataFromAction($action),
+                        $editor,
+                        $submission
+                    );
+                    break;
+            }
+        }
+
+        $reviewRoundDao = DAORegistry::getDAO('ReviewRoundDAO'); /** @var ReviewRoundDAO $reviewRoundDao */
+        $reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /** @var ReviewAssignmentDAO $reviewAssignmentDao */
+        $reviewRoundId = $decision->getData('reviewRoundId');
+
+        $reviewAssignmentDao->deleteByReviewRoundId($reviewRoundId);
+        $reviewRoundDao->deleteById($reviewRoundId);
```

Comment:
Yes, thats the cause . on a clean installation , it will have that `foreign key on delete set to null` as I have updated the migration . For an existing installation, to set that , that update migration first need to run .